eth-contracts/test/governance.test.js

```diff
@@ -0,0 +1,468 @@
+      assert.equal(proposal.signature, signature, 'Expected same proposal.signature')
+      assert.equal(proposal.callData, callData, 'Expected same proposal.callData')
+      assert.equal(proposal.outcome, Outcome.InProgress, 'Expected same outcome')
+      assert.equal(parseInt(proposal.voteMagnitudeYes), 0, 'Expected same voteMagnitudeYes')
+      assert.equal(parseInt(proposal.voteMagnitudeNo), defaultStakeAmount, 'Expected same voteMagnitudeNo')
+      assert.equal(parseInt(proposal.numVotes), 1, 'Expected same numVotes')
+
+      // Confirm all vote states - Vote.No for Voter, Vote.None for all others
+      for (const account of accounts) {
+        const voterVote = await governanceContract.getVoteByProposalAndVoter.call(proposalId, account)
+        if (account == voterAddress) {
+          assert.equal(voterVote, vote)
+        } else {
+          assert.equal(voterVote, defaultVote)
+        }
+      }
+    })
+
+    it('Evaluate successful Proposal + execute Slash', async () => {
+      const proposalId = 1
+      const proposerAddress = stakerAccount1
+      const slashAmount = 1
+      const targetAddress = stakerAccount2
+      const voterAddress = stakerAccount1
+      const vote = Vote.Yes
+      const defaultVote = Vote.None
+      const lastBlock = (await _lib.getLatestBlock(web3)).number
+      const targetContractRegistryKey = delegateManagerKey
+      const targetContractAddress = delegateManagerContract.address
+      const callValue = bigNumberify(0)
+      const signature = 'slash(uint256,address)'
+      const callData = abiEncode(['uint256', 'address'], [slashAmount, targetAddress])
+      const outcome = Outcome.Yes 
+      const txHash = keccak256(
+        abiEncode(
+          ['address', 'uint256', 'string', 'bytes'],
+          [targetContractAddress, callValue, signature, callData]
+        )
+      )
+      const returnData = null
+
+      // Confirm initial Stake state
+      const initialTotalStake = parseInt(await stakingContract.totalStaked())
+      assert.equal(initialTotalStake, defaultStakeAmount * 2)
+      const initialStakeAcct2 = parseInt(await stakingContract.totalStakedFor(targetAddress))
+      assert.equal(initialStakeAcct2, defaultStakeAmount)
+
+      // Call submitProposal + submitProposalVote
+      const submitProposalTxReceipt = await governanceContract.submitProposal(
+        targetContractRegistryKey,
+        callValue,
+        signature,
+        callData,
+        proposalDescription,
+        { from: proposerAddress }
+      )
+      await governanceContract.submitProposalVote(proposalId, vote, { from: voterAddress })
+
+      // Advance blocks to the next valid claim
+      const proposalStartBlockNumber = parseInt(_lib.parseTx(submitProposalTxReceipt).event.args.startBlockNumber)
+      let currentBlockNumber = (await _lib.getLatestBlock(web3)).number
+      while (currentBlockNumber <= proposalStartBlockNumber + votingPeriod) {
```

Comment:
fwiw i have added an 'advanceToBlock' we can use

Reply:
beautiful, done